x-pack/platform/plugins/shared/fleet/server/services/agents/agentless_agent.ts

```diff
@@ -337,6 +356,67 @@ class AgentlessAgentService {
     return response;
   }
 
+  public async listAgentlessDeployments(opts?: { perPage?: number; nextPageToken?: string }) {
+    const logger = appContextService.getLogger();
+
+    const traceId = apm.currentTransaction?.traceparent;
+    const agentlessConfig = appContextService.getConfig()?.agentless;
+    const tlsConfig = this.createTlsConfig(agentlessConfig);
+    const requestConfig: AxiosRequestConfig = {
+      url: prependAgentlessApiBasePathToEndpoint(agentlessConfig, '/deployments'),
+      method: 'GET',
+      params: {
+        next_token: opts?.nextPageToken,
+        limit: opts?.perPage,
+      },
+      ...this.getHeaders(tlsConfig, traceId),
+    };
+
+    const errorMetadata: LogMeta = {
+      trace: {
+        id: traceId,
+      },
+    };
+
+    const requestConfigDebugStatus = this.createRequestConfigDebug(requestConfig);
+
+    logger.debug(`[Agentless API] Start listing agentless deployments`);
+
+    if (!isAgentlessEnabled) {
+      logger.error(
+        '[Agentless API] Agentless API is not supported. Listing agentless deployments is not supported in non-cloud or non-serverless environments'
+      );
+    }
+
+    if (!agentlessConfig) {
+      logger.error('[Agentless API] kibana.yml is currently missing Agentless API configuration');
+    }
+
+    logger.debug(`[Agentless API] Listing agentless deployments with TLS config with certificate`);
+
+    logger.debug(
+      `[Agentless API] Listing agentless deployments with request config ${requestConfigDebugStatus}`
+    );
+
+    const response = await axios(requestConfig).catch((error: AxiosError) => {
+      this.catchAgentlessApiError(
+        'list',
+        error,
+        logger,
+        undefined,
+        requestConfig,
+        requestConfigDebugStatus,
+        errorMetadata,
+        traceId
+      );
+      throw new Error('Unhandled Agentless API list deployments Agentless API', {
```

Comment:
Seems `this.catchAgentlessApiError` has a path for handling unhandled errors. Do we want to lean on that instead, seems this `throw` maybe unreachable?